internal/config/config.go

```diff
@@ -6,26 +6,33 @@ import (
 
 	ghAuth "github.com/cli/go-gh/pkg/auth"
 	ghConfig "github.com/cli/go-gh/pkg/config"
+	"github.com/zalando/go-keyring"
 )
 
 const (
-	hosts   = "hosts"
-	aliases = "aliases"
+	aliases    = "aliases"
+	hosts      = "hosts"
+	oauthToken = "oauth_token"
 )
 
 // This interface describes interacting with some persistent configuration for gh.
 //
 //go:generate moq -rm -out config_mock.go . Config
 type Config interface {
-	AuthToken(string) (string, string)
 	Get(string, string) (string, error)
 	GetOrDefault(string, string) (string, error)
 	Set(string, string, string)
-	UnsetHost(string)
-	Hosts() []string
-	DefaultHost() (string, string)
-	Aliases() *AliasConfig
 	Write() error
+
+	Aliases() *AliasConfig
+	Authentication() *AuthConfig
+
+	// This is deprecated and will be removed, do not use!
+	// Please use cfg.Authentication().Token()
+	AuthToken(string) (string, string)
+	// This is deprecated and will be removed, do not use!
+	// Please use cfg.Authentication().DefaultHost()
+	DefaultHost() (string, string)
```

Comment:
Would have removed these but it would have ballooned the size of this PR. Will do it in a follow up cleanup PR.

Reply:
https://github.com/cli/cli/pull/7049